autoload/clap/provider/search_history.vim

```diff
@@ -0,0 +1,36 @@
+" Author: Mark Wu <markplace@gmail.com>
+" Description: List the search history.
+
+let s:save_cpo = &cpoptions
+set cpoptions&vim
+
+" Derived from fzf.vim
+function! s:search_history() abort
+  let max  = histnr('/')
+  let fmt  = ' %'.len(string(max)).'d '
+  let list = filter(map(range(1, max), 'histget(":", - v:val)'), '!empty(v:val)')
+  return list
+endfunction
+
+function! s:search_history_source() abort
+  let search_hist = s:search_history()
+  let max  = histnr('/')
+  let search_hist_len = len(search_hist)
+  return map(search_hist, 'printf("%4d", search_hist_len - v:key)."  ".v:val')
+endfunction
+
+function! s:search_history_sink(selected) abort
+  let item = matchstr(a:selected, '\d\+\s\+\zs\(.*\)')
+  let @/ = item
+  execute 'normal n'
+endfunction
+
+let s:search_history = {}
+let s:search_history.sink = function('s:search_history_sink')
+let s:search_history.source = function('s:search_history_source')
+let s:search_history.syntax = 'clap_search_history'
```

Comment:
If you use the same syntax file with `clap_search_history`,no need to copy `syntax/clap_command_history`, just set the `syntax` to `clap_command_history`, even though the semantic is not quite right, we don't care as we merely want the highlight.

Reply:
okay.

Reply:
Done!